.ci/Jenkinsfile

```diff
@@ -28,70 +28,107 @@ pipeline {
         }
-      }
-      post {
-        always {
-          dir("${BASE_DIR}") {
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/test-results/*.xml')
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/kubectl-dump.txt')
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/elastic-stack-dump/stack/logs/*.log')
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/elastic-stack-dump/check/logs/*.log')
-            junit(allowEmptyResults: false,
-                keepLongStdio: true,
-                testResults: "build/test-results/*.xml")
-            coverageReport('build/test-coverage')
+        /**
+        Run the unit tests suite
+        */
+        stage('Unit tests') {
+          steps {
+            cleanup()
+            withMageEnv(){
+              dir("${BASE_DIR}"){
+                sh(label: 'Check',script: 'make test-go-ci')
+              }
+            }
+          }
+          post {
+            always {
+              dir("${BASE_DIR}") {
+                archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/test-results/*.xml')
+                junit(allowEmptyResults: false,
+                    keepLongStdio: true,
+                    testResults: "build/test-results/*.xml")
+                coverageReport('build/test-coverage')
+              }
+            }
           }
         }
-      }
-    }
-    stage('Release') {
-      when {
-        tag pattern: '(v)?\\d+\\.\\d+\\.\\d+', comparator: 'REGEXP'
-      }
-      steps {
-        withMageEnv(){
-          withCredentials([string(credentialsId: "${GITHUB_TOKEN_CREDENTIALS}", variable: 'GITHUB_TOKEN')]) {
-            dir("${BASE_DIR}") {
-              sh 'curl -sL https://git.io/goreleaser | bash'
+        stage('Integration Tests') {
+          failFast true
+          options { skipDefaultCheckout() }
+          steps {
+            withGithubNotify(context: 'Integration Tests', tab: 'tests') {
+              cleanup()
+              dir("${BASE_DIR}") {
+                script {
+                  def artifacts = [:]
+                  def testCommands = ['test-stack-command', 'test-check-packages', 'test-profiles-command']
+
+                  artifacts['test-stack-command'] = [
+                    'build/elastic-stack-dump/**/*.log'
+                  ]
+                  artifacts['test-check-packages'] = [
+                    'build/test-results/*.xml', 'build/kubectl-dump.txt', 'build/elastic-stack-dump/**/*.log'
+                  ]
+                  artifacts['test-profiles-command'] = []
+
+                  def parallelTasks = [:]
+
+                  testCommands.each { testCommand ->
+                    def junitArtifacts = false
+                    def publishCoverage = false
+
+                    if (testCommand == 'test-check-packages') {
+                      junitArtifacts = true
+                      publishCoverage = true
+                    }
```

Comment:
Making explicit what each command is doing in terms of publishing CI artifacts. I was tempted to create a separate file, but I think here the code is simpler to read/maintain.